sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java

```diff
@@ -886,16 +902,13 @@ void initializeHttpPipelineAdapter() {
         // as they should directly be set on the pipeline.
         HttpPipeline httpPipeline = options.getHttpPipeline();
         if (httpPipeline != null) {
+            this.httpPipeline = httpPipeline;
             httpPipelineAdapter = new HttpPipelineAdapter(httpPipeline, options);
         } else {
             // If http client is set on the credential, then it should override the proxy options if any configured.
             HttpClient httpClient = options.getHttpClient();
-            if (httpClient != null) {
-                httpPipelineAdapter = new HttpPipelineAdapter(setupPipeline(httpClient, this.options), options);
-            } else if (options.getProxyOptions() == null) {
-                //Http Client is null, proxy options are not set, use the default client and build the pipeline.
-                httpPipelineAdapter = new HttpPipelineAdapter(setupPipeline(HttpClient.createDefault(), options), options);
-            }
+            this.httpPipeline = setupPipeline(httpClient == null ? HttpClient.createDefault() : httpClient);
```

Comment:
this breaks the proxy flow I think (backward compat), the Pipeline adapter in current flow is only configured if either httpClient is not null or proxy options are not configured.

With this change, the pipeline adapter will get configured if httpClient is null and proxy options are configured.

Reply:
This call will break with this change. With proxy options (backward compat), this flow is currently invoked.

https://github.com/Azure/azure-sdk-for-java/pull/40722/files#diff-b5937fc0e636a1c6954f1b77ed7149066d1f7048dd85d8db025609d2615c514cR277

To retain this, I'd say, we move this logic up, and before setting client Assertion credentials, check if pipeline is null then create the pipeline there for the assertion to consume.

Reply:
good catch, fixing!